### PR TITLE
Feature/spiral strategy

### PR DESCRIPTION
**Pull request by Sporarum**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/searcharea/SearchArea.kt [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+//    abstract fun reset()
+//
+//    fun getLatLng(): MutableLiveData<MutableList<LatLng>> {
+//        return latLngs
+//    }
+//
+//    fun getAdditionalProps(): MutableLiveData<MutableMap<String, Double>> {
+//        return props
+//    }
+}
```
Comment (Nchappui) [inline app/src/main/java/ch/epfl/sdp/searcharea/SearchArea.kt:22]: Are all these commented methods usefull to keep?

File: build.gradle [hunk 2]
```diff
@@ -25,6 +25,9 @@ allprojects {
         maven {
             url "https://dl.bintray.com/videolan/Android"
         }
+        maven {
+            url "https://jitpack.io"
+        }
```
Comment (Nchappui) [inline build.gradle:30]: I don't get what it is for, can you please add a comment

File: app/src/main/java/ch/epfl/sdp/searcharea/SearchArea.kt (hunk 1 above)
Comment (Sporarum) [reply app/src/main/java/ch/epfl/sdp/searcharea/SearchArea.kt:22]: Nope, I removed them in local, waiting for Cirrus before pushing

File: build.gradle (hunk 2 above)
Comment (Sporarum) [reply build.gradle:30]: It's to add the util library for the map
(this one: https://github.com/mastrgamr/mapbox-android-utils/blob/master/library/src/main/java/net/mastrgamr/mbmapboxutils/SphericalUtil.java )

File: app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            "The maximum distance between passes must be strictly positive"
+        }
+        this.maxDistBetweenLines = maxDistBetweenLinesIn
+    }
+
+    override fun acceptArea(searchArea: SearchArea): Boolean {
+        return searchArea is CircleArea
+    }
+
+    override fun createFlightPath(startingPoint: LatLng, searchArea: SearchArea): List<LatLng> {
```
[the remark this answers is not in the document]
Comment (Ph0tonic) [reply app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt:32]: Can easily be fixed.

File: app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        require(acceptArea(searchArea)) { "This strategy does not accept this type of area" }
+        val area = searchArea as CircleArea
+        val center = area.center
+        val outer = area.outer
+        val radius = center.distanceTo(outer)
+
+        val path = ArrayList<LatLng>()
+
+        val earthRadius = 6378137
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt:41]: Extract to a companion object.

File: build.gradle [hunk 5]
```diff
@@ -25,6 +25,9 @@ allprojects {
         maven {
             url "https://dl.bintray.com/videolan/Android"
         }
+        maven {
+            url "https://jitpack.io"
```
Comment (Ph0tonic) [inline build.gradle:29]: Coul you merge those two maven tags?

Comment (Sporarum) [reply build.gradle:29]: Sadly not, they each have one field url, not a list of urls

File: app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt [hunk 6]
```diff
@@ -18,7 +18,7 @@ import kotlin.math.sqrt
 class SpiralStrategy(maxDistBetweenLinesIn: Double) : OverflightStrategy {
     private val maxDistBetweenLines: Double
 
-    companion object{
+    companion object {
         const val earthRadius = 6378137
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt:22]: Make it private and const.

File: app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt [hunk 7]
```diff
@@ -39,14 +39,19 @@ class SpiralStrategy(maxDistBetweenLinesIn: Double) : OverflightStrategy {
         val center = area.center
         val outer = area.outer
         val radius = center.distanceTo(outer)
+        val angleToOuter = Math.toRadians(computeHeading(center, outer))
+
+        return generatePointsAlongSpiral(center, radius, angleToOuter)
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt:44]: What is this return for?

Comment (Sporarum) [reply app/src/main/java/ch/epfl/sdp/mission/SpiralStrategy.kt:44]: Because createFlightPath returns a list of LatLng

**Pull request merged**